aws-api/src/main/java/com/amplifyframework/api/aws/AWSApiPlugin.java

```diff
@@ -736,4 +753,53 @@ private void transitionTo(ApiEndpointStatus newStatus) {
             }
         }
     }
+
+    /**
+     * Builds an {@link AWSApiPlugin}.
+     */
+    public static final class Builder {
+        private ApiAuthProviders apiAuthProviders;
+        private final Map<String, OkHttpConfigurator> apiConfigurators;
+
+        private Builder() {
+            this.apiAuthProviders = ApiAuthProviders.noProviderOverrides();
+            this.apiConfigurators = new HashMap<>();
+        }
+
+        /**
+         * Specify authentication providers.
+         * @param apiAuthProviders A set of authentication providers to use for API calls
+         * @return Current builder instance, for fluent construction of plugin
+         */
+        @NonNull
+        public Builder apiAuthProviders(@NonNull ApiAuthProviders apiAuthProviders) {
+            Objects.requireNonNull(apiAuthProviders);
+            Builder.this.apiAuthProviders = apiAuthProviders;
+            return Builder.this;
+        }
+
+        /**
+         * Apply customizations to an underlying OkHttpClient that will be used
+         * for a particular API.
+         * @param forApiName The name of the API for which these customizations should apply
```

Comment:
```suggestion
         * @param forApiName The name of the API for which these customizations should apply.  This can be found in your `amplifyconfiguration.json` file.
```